introduction_to_amazon_algorithms/xgboost_abalone/xgboost_inferenece_script_mode.ipynb

```diff
@@ -0,0 +1,448 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Use Your Own Inference Code with Amazon SageMaker XGBoost Algorithm\n",
+    "_**Customized inference for computing SHAP values with Amazon SageMaker XGBoost script mode**_\n",
+    "\n",
+    "---\n",
+    "\n",
+    "## Contents\n",
+    "1. [Introduction](#Introduction)\n",
+    "2. [Setup](#Setup)\n",
+    "3. [Training the XGBoost model](#Training-the-XGBoost-model)\n",
+    "4. [Deploying the XGBoost endpoint](#Deploying-the-XGBoost-endpoint)\n",
+    "\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "\n",
+    "This notebook shows how you can configure the SageMaker XGBoost model server by defining `input_fn`, `predict_fn`, and `output_fn` (or `transform_fn`) functions in the Python source file you pass to the XGBoost constructor in the SageMaker Python SDK. We will write a customized inference script that is designed to illustrate how [SHAP](https://github.com/slundberg/shap) values enable the interpretion of XGBoost models.\n",
```

Comment:
Can we give a very brief explanation of what these functions do? Something like:

"...functions to customize how your model pre- and post- processes data and how it serves predictions."